### ISSUE TITLE
French exclamation mar' key not recognized (and solution)

### ISSUE THREAD

**Issue by Yomgui33** [david-janssen/kmonad#192]
After scratching my head for a while to understand why I kept getting an error each time I pressed the ! key on my azerty ISO (same position as the / on most international keyboards), I finally found a solution.

The keycode for this key is different from the / key and is not handled by default by kmonad. Instead of being VK_OEM_2 like on other ISO keyboards, its code is VK_OEM_8 which is commented out in the source.

The solution was to uncomment it in **\src\KMonad\Keyboard\IO\Windows\Type.hs** and assign a name to it:

`, (0xDF, KeyExclamation)   -- Defined ask VK_OEM_8`

Then add this name to **src\KMonad\Keyboard\Keycode.hs**

After compiling, the error was gone. This might be useful to someone else.

Any reason why VK_OEM_8 is commented out by default?

Also: amazing program. Thanks for creating it! Home row mods are now working on my non QWK Realforce.

**Comment by david-janssen** [comment]
Ahhh, this is cool. Did you make a pull-request?

Sorry for the late reply, been a bit under the weather. 

The reason it was commented out is because I did the Windows support myself, but I hardly use windows, and I had no idea what the keycode was.

**Comment by Yomgui33** [comment]
No, sorry, no idea how github really works, apart from downloading releases. In fact, it's a miracle I managed to compile the source myself, as I had zero knowledge of Haskell. :)

Just wanted to leave the workaround somewhere, in case another french user stumbles upon the same issue.

**Comment by david-janssen** [comment]
Very good.

I'm working on a complete keycode reorganization anyway to ease cross-platform support, so once that's done, hopefully this will be ironed out too.

Kudos on compiling haskell by yourself! I remember my struggles when I first started out :-).

Going to leave this open until the keycode-fix is in, as a reference for fellow french friends.
